components/x-teaser-timeline/__tests__/lib/transform.test.js

```diff
@@ -298,5 +298,55 @@ describe('buildModel', () => {
 				groupedItems[2]
 			])
 		})
+		test('returns correct model for multiple custom slots', () => {
+			const result = buildModel({
+				items,
+				timezoneOffset: 0,
+				localTodayDate: '2020-01-14',
+				customSlotContent: [{ foo: 1 }, { bar: 2 }],
+				customSlotPosition: [0, 3]
+			})
+			expect(result).toEqual([
+				{
+					...groupedItems[0],
+					items: [
+						{ foo: 1 },
+						groupedItems[0].items[0],
+						groupedItems[0].items[1],
+						groupedItems[0].items[2],
+						{ bar: 2 },
+						groupedItems[0].items[3]
+					]
+				},
+				groupedItems[1],
+				groupedItems[2]
+			])
+		})
+		test('returns correct model for multiple custom slots off end of all groups', () => {
+			const result = buildModel({
+				items,
+				timezoneOffset: 0,
+				localTodayDate: '2020-01-14',
+				customSlotContent: [{ foo: 1 }, { bar: 2 }],
+				customSlotPosition: [0, 10]
```

Comment:
I think it would be worth having a test case where the first insert position isn't 0, to validate the offset

Reply:
I've added another test [2,3] to show the effects at a non-0 position.